server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -902,7 +902,11 @@ public boolean shouldUseConcurrentSearch() {
      * Evaluate if parsed request supports concurrent segment search
      */
     public void evaluateRequestShouldUseConcurrentSearch() {
-        if (sort != null && sort.isSortOnTimeSeriesField()) {
+        // Do not use concurrent segment search for system indices or throttled requests. See:
+        // https://github.com/opensearch-project/OpenSearch/issues/12951
+        if (indexShard.isSystem() || indexShard.indexSettings().isSearchThrottled()) {
```

Comment:
How about if we move these checks to `evaluateConcurrentSegmentSearchSettings` as these are index settings and metadata related checks ?

Reply:
This makes sense to me, will make this change in the next revision!